mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingLanguageDao.java

```diff
@@ -23,6 +35,42 @@ default HibLanguage create(String languageName, String languageTag, String uuid)
 		return language;
 	}
 
+	@Override
+	default void assign(HibLanguage language, HibProject project, EventQueueBatch batch, boolean throwOnExisting) {
+		project.getLanguages().stream()
+			.filter(l -> l.getLanguageTag().equals(language.getLanguageTag()))
+			.findAny()
+			.ifPresentOrElse(existing -> {
+					if (throwOnExisting) {
+						throw error(BAD_REQUEST, "error_language_already_assigned", language.getLanguageTag(), project.getName());
```

Comment:
Isn't CONFLICT the better response code here?
Or should we remove this, because throwOnExisting is currently set to false on all occasions?

Reply:
Same as above. I used to treat BAD_REQUEST as something incorrect coming from the client input, though correctly processed by the server, which fits these cases fairly well.